lib/puma/server.rb

```diff
@@ -476,19 +480,19 @@ def process_client(client)
 
             requests += 1
 
-            # As an optimization, try to read the next request from the
-            # socket for a short time before returning to the reactor.
-            fast_check = @status == :run
+            Thread.pass
```

Comment:
> hoping the reactor is scheduled to run

Yes, that was the reason.  Also, it might be a tough thing to verify either way.  Maybe leave as is with a better comment is the best solution?

Reply:
> Maybe leave as is with a better comment is the best solution?

I think a comment is fine. I'm just thinking of long term maintenance. So if someone wants to modify the code there's a hint as to whats safe or desirable. Here's my stab at some wording:

> In this code path the keepalive request may be put back into the reactor. Ideally we want to serve requests in the order they came in, even when those requests are re-using a connection. This `Thread.pass` is an attempt to hint to the OS that the reactor can read in a new connection from the TCP socket which might have been sent after the currently processing request, but before the next request in its connection.

Reply:
LGTM.  I wanted to test if the statement made any difference, but with MRI, JRuby & TruffleRuby, it might involve a lot of testing.

This weekend I looked at stats when using workers to see how well requests were distributed amongst the workers.  As an example, with four workers, perfect distribution would be 25% for each worker.  I think the worst I saw was a +/-15% diff, so lowest was 21.25%, highest was 28.75%.  That's from memory, many runs were tighter than that.

Reply:
I'm confused why this is here even reading the above comments. What is the condition where you expect the reactor to be running simultaniously?